tests/lib/scope-analysis.js

```diff
@@ -0,0 +1,355 @@
+        const code = `
+type foo = any;
+function bar(foo: any) {}
+`;
+        const config = {
+            parser: "typescript-eslint-parser",
+            rules: {
+                "no-shadow": "error"
+            }
+        };
+        const messages = linter.verify(code, config, { filename: "issue.ts" });
+
+        assert.deepStrictEqual(messages, []);
+    });
+
+    test("https://github.com/eslint/typescript-eslint-parser/issues/466", () => {
+        const code = `
+/*globals document, selector */
+const links = document.querySelectorAll( selector ) as NodeListOf<HTMLElement>
+`;
+        const config = {
+            parser: "typescript-eslint-parser",
+            rules: {
+                "no-undef": "error"
+            }
+        };
+        const messages = linter.verify(code, config, { filename: "issue.ts" });
+
+        assert.deepStrictEqual(messages, []);
+    });
+
+    test("https://github.com/eslint/typescript-eslint-parser/issues/471", () => {
+        const code = `
+class X {
+  field = {}
+}
+`;
+        const config = {
+            parser: "typescript-eslint-parser",
+            rules: {
+                "no-undef": "error"
+            }
+        };
+        const messages = linter.verify(code, config, { filename: "issue.ts" });
+
+        assert.deepStrictEqual(messages, []);
+    });
+
+    test("https://github.com/eslint/typescript-eslint-parser/issues/487", () => {
+        const code = `
+export default class Test {
+    private status: string;
+    getStatus() {
+        return this.status;
+    }
+}
+`;
+        const config = {
+            parser: "typescript-eslint-parser",
+            rules: {
+                "no-restricted-globals": ["error", "status"]
+            }
+        };
+        const messages = linter.verify(code, config, { filename: "issue.ts" });
+
+        assert.deepStrictEqual(messages, []);
+    });
+
+    test("https://github.com/eslint/typescript-eslint-parser/issues/535", () => {
+        const code = `
+function foo({ bar }: { bar: string }) {
+    console.log(bar);
+}
+`;
+        const config = {
+            parser: "typescript-eslint-parser",
+            rules: {
+                "no-dupe-args": "error",
+                "no-redeclare": "error"
+            }
+        };
+        const messages = linter.verify(code, config, { filename: "issue.ts" });
+
+        assert.deepStrictEqual(messages, []);
+    });
+
+    test("https://github.com/eslint/typescript-eslint-parser/issues/535", () => {
+        const code = `
+import {
+  observable,
+} from 'mobx';
+
+export default class ListModalStore {
+  @observable
+  orderList: IObservableArray<BizPurchaseOrderTO> = observable([]);
+}
+`;
+        const config = {
+            parser: "typescript-eslint-parser",
+            rules: {
+                "no-unused-vars": "error"
+            }
+        };
+        const messages = linter.verify(code, config, { filename: "issue.ts" });
+
+        assert.deepStrictEqual(messages, []);
```

Comment:
How come you are frequently using `assert` rather than jest's `expect()` within the jest `test()` blocks?

I can't say I have ever seen or done this so just calling it out in case there is any chance it is masking some issues/causing false positives...

Reply:
Oops. I have used `mocha` always, so I couldn't have the idea that `jest` has assertion functions. I will fix it.